File: java/cours/src/session3/TD/MotIterableAvecIterateurEtVisiteur/MotNonVide.java
class MotNonVide implements Mot {
	private char element;
	private Mot reste;
	private int taille;

	public MotNonVide(char i, Mot ens) {
		this.element = i;
		this.reste = ens;
		this.taille = 1 + ens.taille();
	}

	@Override
	public boolean estMotNonVide() {
		return true;
	}

	@Override
	public int taille() {
		return this.taille;
	}

	@Override
	public char element() {
		return this.element;
	}

	@Override
	public Mot reste() {
		return this.reste;
	}
}

File: typescript/session3/MotIterableAvecIterateurEtVisiteur/MotNonVide.ts
import { Mot } from "./Mot";
import { Char } from "../Char";


export class MotNonVide extends Mot {

  private premiereLettre: Char;
  private resteMot: Mot;
  private tailleMot: number;

  public constructor(i: Char, ens: Mot) {
    super();
    this.premiereLettre = i;
    this.resteMot = ens;
    this.tailleMot = 1 + ens.taille();
  }

  public taille(): number {
    return this.tailleMot;
  }

  public estMotNonVide(): boolean {
    return true;
  }

  public element(): Char {
    return this.premiereLettre;
  }

  public reste(): Mot {
    return this.resteMot;
  }
}